ambry-router/src/main/java/com.github.ambry.router/PutOperation.java

```diff
@@ -1604,12 +1612,21 @@ private void finalizeMetadataChunk() {
               passedInBlobProperties.isEncrypted(), passedInBlobProperties.getExternalAssetTag());
       if (isStitchOperation() || getNumDataChunks() > 1) {
         // values returned are in the right order as TreeMap returns them in key-order.
-        List<StoreKey> orderedChunkIdList = new ArrayList<>(indexToChunkIds.values());
-        buf = MetadataContentSerDe.serializeMetadataContent(intermediateChunkSize, getBlobSize(), orderedChunkIdList);
+        List<Pair<StoreKey, Long>> orderedChunkIdList = new ArrayList<>(indexToChunkIdsAndChunkSizes.values());
```

Comment:
Move this to inside the else if (line 1620) so that we don't have to construct it for v2

Reply:
done